xl/player/player.py

```diff
@@ -187,15 +195,48 @@ def play(self, track, start_at=None, paused=False):
         if track is None:
             self.stop()
         else:
+            self.stop()
             if self.is_stopped():
                 event.log_event('playback_player_start', self, track)
 
-            play_args = self._get_play_params(track, start_at, paused, False)
-            self._engine.play(*play_args)
-            if play_args[2]:
+            self._current_play_args = self._get_play_params(
+                track, start_at, paused, False
+            )
+
+            self._delayed_until = 0
+            self._cancel_delayed_stop()
+            self._track = track
+            start_at = self._current_play_args[1]
+
+            if start_at < 0:
+                logger.debug('entering delayed start')
+                self._delayed_until = -1 * start_at + time.time()
```

Comment:
If you just had the timeout call you at the correct time, this wouldn't be needed?